src/azure-cli-core/azure/cli/core/breaking_change.py

```diff
@@ -0,0 +1,285 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import abc
+import enum
+import re
+
+import packaging.version
+
+import requests
+
+from azure.cli.core import __version__
+
+NEXT_BREAKING_CHANGE_RELEASE = '2.61.0'
+
+
+def _next_breaking_change_version_from_milestone(cur_version):
+    owner = "Azure"
+    repo = "azure-cli"
+    # The GitHub API v3 URL for milestones
+    url = f"https://api.github.com/repos/{owner}/{repo}/milestones"
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        milestones = response.json()
+    except requests.RequestException as e:
+        return None
+    for milestone in milestones:
+        try:
+            if 'breaking change' in milestone['title'].lower():
+                pattern = re.compile(r'Azure CLI version: *(?P<version>[\d.]+) *$', re.MULTILINE | re.IGNORECASE)
+                match = re.search(pattern, milestone['description'])
+                if match:
+                    version = match.group('version')
+                    parsed_version = packaging.version.parse(version)
+                    if parsed_version > cur_version:
+                        return version
+        except (IndexError, KeyError) as e:
+            pass
+    return None
+
+
+__bc_version = None
+
+
+def _next_breaking_change_version():
+    global __bc_version
+    if __bc_version:
+        return __bc_version
+    cur_version = packaging.version.parse(__version__)
+    next_bc_version = packaging.version.parse(NEXT_BREAKING_CHANGE_RELEASE)
+    if cur_version >= next_bc_version:
+        fetched_bc_version = _next_breaking_change_version_from_milestone(cur_version)
+        if fetched_bc_version:
+            __bc_version = fetched_bc_version
+            return fetched_bc_version
+    __bc_version = NEXT_BREAKING_CHANGE_RELEASE
+    return NEXT_BREAKING_CHANGE_RELEASE
```

Comment:
When we release a CLI version with NEXT_BREAKING_CHANGE_RELEASE forget to change, when the cli users will see a different version every half year. My recommendation is that change the NEXT_BREAKING_CHANGE_RELEASE version number in the CLI release pipeline. Every time we bump up the cli version, the pipeline will check and change the NEXT_BREAKING_CHANGE_RELEASE version.

Reply:
I agree with that. If we could update the `NEXT_BREAKING_CHANGE_RELEASE` after breaking change release in pipeline, we don't need a fallback logic to get the version number.